web/src/queries/software.ts

```diff
@@ -166,6 +187,22 @@ const useRegisterMutation = () => {
   return useMutation(query);
 };
 
+/**
+ * Hook that builds a mutation for registering an addon
+ *
+ */
+const useRegisterAddonMutation = () => {
+  const queryClient = useQueryClient();
+
+  const query = {
+    mutationFn: registerAddon,
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ["software/registration/addons/registered"] });
```

Comment:
NP: not because of your changes, but something I'd like to do/improve is how we use query keys to avoid mistyping them and having bugs. Here you could go for `queryKey: registeredAddonsQuery().queryKey`. Or we could have constants for queryKeys. I don't know, but we have to found a way. Out of the scope of this PR, but adding the comment for making others aware. @imobachgs, @joseivanlopez friendly ping since you both had worked with queries quite a lot.